File: Singleton.java
package com.example.kotlincodes;

public class Singleton{
    private static Singleton INSTANCE;
    private Singleton(){}
    public static Singleton getInstance(){
        if (INSTANCE == null) {                // Single Checked
            synchronized (Singleton.class) {
                if (INSTANCE == null) {        // Double checked
                    INSTANCE = new Singleton();
                }
            }
        }
        return INSTANCE;
    }
    private int count = 0;
    public int count(){ return count++; }
}

File: Singleton.kt
package com.example.kotlincodes

// Singleton class ensuring only one instance exists
object Singleton {
    private var count = 0

    fun increment() {
        count++
        println("Current count: $count")
    }
}

fun main() {
    // Accessing Singleton instance and calling its function
    Singleton.increment()
    Singleton.increment()
}
